test-cargo-miri/run-test.py

```diff
@@ -42,13 +42,16 @@ def test(name, cmd, stdout_ref, stderr_ref, stdin=b'', env={}):
     (stdout, stderr) = p.communicate(input=stdin)
     stdout = stdout.decode("UTF-8")
     stderr = stderr.decode("UTF-8")
-    if p.returncode == 0 and normalize_stdout(stdout) == open(stdout_ref).read() and stderr == open(stderr_ref).read():
+    stdout = normalize_stdout(stdout)
+    expected_stdout = open(stdout_ref).read()
+    if p.returncode == 0 and stdout == expected_stdout and stderr == open(stderr_ref).read():
         # All good!
         return
     # Show output
-    print("Test stdout or stderr did not match reference!")
+    print(f"{stdout_ref} or {stderr_ref} did not match reference!")
     print("--- BEGIN test stdout ---")
-    print(stdout, end="")
+    for text in difflib.unified_diff(expected_stdout.split("\n"), stdout.split("\n")):
+        print(text)
```

Comment:
I assume this prints a diff now? Then please adjust the BEGIN and END messages before and after. And also do this for stderr.